aries_cloudagent/messaging/jsonld/routes.py

```diff
@@ -42,34 +41,36 @@ async def sign(request: web.BaseRequest):
         request: aiohttp request object
 
     """
-    response = {}
     try:
         context: AdminRequestContext = request["context"]
+        session = await context.session()
         body = await request.json()
         verkey = body.get("verkey")
         doc = body.get("doc")
-        credential = doc["credential"]
-        signature_options = doc["options"]
-
-        async with context.session() as session:
-            wallet = session.inject(BaseWallet, required=False)
-            if not wallet:
-                raise web.HTTPForbidden(reason="No wallet available")
-            document_with_proof = await sign_credential(
-                credential, signature_options, verkey, wallet
-            )
-
-        response["signed_doc"] = document_with_proof
-    except (DroppedAttributeError, MissingVerificationMethodError) as err:
-        response["error"] = str(err)
-
-    return web.json_response(response)
+        credential = doc.get("credential")
+        signature_options = doc.get("options")
+        doc_with_proof = await sign_credential(
+            session, credential, signature_options, verkey
+        )
+    except (DIDError, ResolverError, WalletError, InjectionError) as err:
+        if isinstance(err, BaseError):
+            raise web.HTTPBadRequest(reason=err.roll_up) from err
+        # TODO: PyDID errors should be wrapped in did resolver classes to support .roll_up
+        else:
+            raise web.HTTPBadRequest(
+                reason=f"internal raised error: '{repr(err)}'"
+            ) from err
+    return web.json_response({"signed_doc": doc_with_proof})
```

Comment:
the changes seen here are from sign_credintial being updated to take a session, using getters instead of brackets to access dictionary values, updated error handling, and inline returned value. I think you are right with `DIDError, ResolverError, InjectionError` being residual resolver code. 
What other changes would you like to revert back to? We could update the schema to remove the benefits of using getters.

Reply:
Our changes here are altering the API by returning an error instead of just setting the error in the response. I'm not sure that's the right choice but, for this change set, I think it would be better to just leave it as it was, minus the changes to passing a session instead of passing a wallet.